docs/api/navigation-history.md

```diff
@@ -74,3 +74,20 @@ Returns `boolean` - Whether the navigation entry was removed from the webContent
 #### `navigationHistory.getAllEntries()`
 
 Returns [`NavigationEntry[]`](structures/navigation-entry.md) - WebContents complete history.
+
+#### `navigationHistory.restore(index, entries)`
+
+Restores navigation history. Will make a best effort to restore not just the navigation stack but
+also the state of the individual pages - for instance including HTML form values or the scroll
+position. It's recommended to call this API before any navigation entries are created, so ideally
+before you call `loadURL()` or `loadFile()` on the `webContents` object.
+
+This API allows you to create common flows that aim to restore, recreate, or clone other webContents.
+
+* `offset` Integer
+* `entries` [NavigationEntry[]](structures/navigation-entry.md)
+
+Returns `Promise<void>` - the promise will resolve when the page has finished loading the selected navigation entry
+(see [`did-finish-load`](web-contents.md#event-did-finish-load)), and rejects
+if the page fails to load (see
+[`did-fail-load`](web-contents.md#event-did-fail-load)). A noop rejection handler is already attached, which avoids unhandled rejection errors.
```

Comment:
> A noop rejection handler is already attached, which avoids unhandled rejection errors.

Question: how do we feel about this behavior for `loadURL`? To me, it seems like a questionable feature of that API that makes it easier to miss the fact that a view failed to load. It's just hard for us to remove that behavior without causing a lot of noise. I wonder if we should *not* cary that behavior over into `navigationHistory.restore`?

Reply:
We talked in Slack: We don't like it and we might change that after this PR in both APIs.